### PR TITLE
Update doc: Arduino as IDF component; fixes #5833

### LINKED ISSUE #5833: Arduino as a ESP-IDF component Build Failure
### Hardware:
Board:							ESP32-WROOVER-IE
Core Installation version:			ESP-IDF v5.0-dev-79-gbcbef9a8db
IDE name:						ESP-IDF CLI
Computer OS:                                         Debian Bullseye running under WSL2

### Description:
Trying to complete the [getting started instructions](https://docs.espressif.com/projects/arduino-esp32/en/latest/esp-idf_component.html#installation).

### Sketch: 
```cpp

#include "Arduino.h"

void setup(){
    Serial.begin(115200);
}

void loop(){
    Serial.println("loop");
    delay(1000);
}
```

### Debug Messages:
```
[1019/1229] Building C object esp-idf/arduino-esp32/CMakeFiles/__idf_arduino-esp32.dir/cores/esp32/esp32-hal-timer.c.obj
FAILED: esp-idf/arduino-esp32/CMakeFiles/__idf_arduino-esp32.dir/cores/esp32/esp32-hal-timer.c.obj
/home/nthederahn/.espressif/tools/xtensa-esp32-elf/esp-2021r2-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc -DHAVE_CONFIG_H -DMBEDTLS_CONFIG_FILE=\"mbedtls/esp_config.h\" -Iconfig -I../components/arduino-esp32/variants/esp32 -I../components/arduino-esp32/cores/esp32 -I../components/arduino-esp32/libraries/ArduinoOTA/src -I../components/arduino-esp32/libraries/AsyncUDP/src -I../components/arduino-esp32/libraries/BLE/src -I../components/arduino-esp32/libraries/BluetoothSerial/src -I../components/arduino-esp32/libraries/DNSServer/src -I../components/arduino-esp32/libraries/EEPROM/src -I../components/arduino-esp32/libraries/ESP32/src -I../components/arduino-esp32/libraries/ESPmDNS/src -I../components/arduino-esp32/libraries/FFat/src -I../components/arduino-esp32/libraries/FS/src -I../components/arduino-esp32/libraries/HTTPClient/src -I../components/arduino-esp32/libraries/HTTPUpdate/src -I../components/arduino-esp32/libraries/LittleFS/src -I../components/arduino-esp32/libraries/NetBIOS/src -I../components/arduino-esp32/libraries/Preferences/src -I../components/arduino-esp32/libraries/RainMaker/src -I../components/arduino-esp32/libraries/SD_MMC/src -I../components/arduino-esp32/libraries/SD/src -I../components/arduino-esp32/libraries/SimpleBLE/src -I../components/arduino-esp32/libraries/SPIFFS/src -I../components/arduino-esp32/libraries/SPI/src -I../components/arduino-esp32/libraries/Ticker/src -I../components/arduino-esp32/libraries/Update/src -I../components/arduino-esp32/libraries/USB/src -I../components/arduino-esp32/libraries/WebServer/src -I../components/arduino-esp32/libraries/WiFiClientSecure/src -I../components/arduino-esp32/libraries/WiFi/src -I../components/arduino-esp32/libraries/WiFiProv/src -I../components/arduino-esp32/libraries/Wire/src -I../components/arduino-esp32/cores/esp32/libb64 -I/home/nthederahn/esp/esp-idf/components/newlib/platform_include -I/home/nthederahn/esp/esp-idf/components/freertos/include -I/home/nthederahn/esp/esp-idf/components/freertos/include/esp_additions/freertos -I/home/nthederahn/esp/esp-idf/components/freertos/port/xtensa/include -I/home/nthederahn/esp/esp-idf/components/freertos/include/esp_additions -I/home/nthederahn/esp/esp-idf/components/esp_hw_support/include -I/home/nthederahn/esp/esp-idf/components/esp_hw_support/include/soc -I/home/nthederahn/esp/esp-idf/components/esp_hw_support/include/soc/esp32 -I/home/nthederahn/esp/esp-idf/components/esp_hw_support/port/esp32/. -I/home/nthederahn/esp/esp-idf/components/heap/include -I/home/nthederahn/esp/esp-idf/components/log/include -I/home/nthederahn/esp/esp-idf/components/lwip/include/apps -I/home/nthederahn/esp/esp-idf/components/lwip/include/apps/sntp -I/home/nthederahn/esp/esp-idf/components/lwip/lwip/src/include -I/home/nthederahn/esp/esp-idf/components/lwip/port/esp32/include -I/home/nthederahn/esp/esp-idf/components/lwip/port/esp32/include/arch -I/home/nthederahn/esp/esp-idf/components/soc/include -I/home/nthederahn/esp/esp-idf/components/soc/esp32/. -I/home/nthederahn/esp/esp-idf/components/soc/esp32/include -I/home/nthederahn/esp/esp-idf/components/hal/esp32/include -I/home/nthederahn/esp/esp-idf/components/hal/include -I/home/nthederahn/esp/esp-idf/components/hal/platform_port/include -I/home/nthederahn/esp/esp-idf/components/esp_rom/include -I/home/nthederahn/esp/esp-idf/components/esp_rom/include/esp32 -I/home/nthederahn/esp/esp-idf/components/esp_rom/esp32 -I/home/nthederahn/esp/esp-idf/components/esp_common/include -I/home/nthederahn/esp/esp-idf/components/esp_system/include -I/home/nthederahn/esp/esp-idf/components/esp_system/port/soc -I/home/nthederahn/esp/esp-idf/components/esp_system/port/public_compat -I/home/nthederahn/esp/esp-idf/components/esp32/include -I/home/nthederahn/esp/esp-idf/components/xtensa/include -I/home/nthederahn/esp/esp-idf/components/xtensa/esp32/include -I/home/nthederahn/esp/esp-idf/components/driver/include -I/home/nthederahn/esp/esp-idf/components/driver/esp32/include -I/home/nthederahn/esp/esp-idf/components/esp_pm/include -I/home/nthederahn/esp/esp-idf/components/esp_ringbuf/include -I/home/nthederahn/esp/esp-idf/components/efuse/include -I/home/nthederahn/esp/esp-idf/components/efuse/esp32/include -I/home/nthederahn/esp/esp-idf/components/vfs/include -I/home/nthederahn/esp/esp-idf/components/esp_wifi/include -I/home/nthederahn/esp/esp-idf/components/esp_event/include -I/home/nthederahn/esp/esp-idf/components/esp_netif/include -I/home/nthederahn/esp/esp-idf/components/esp_eth/include -I/home/nthederahn/esp/esp-idf/components/tcpip_adapter/include -I/home/nthederahn/esp/esp-idf/components/esp_phy/include -I/home/nthederahn/esp/esp-idf/components/esp_phy/esp32/include -I/home/nthederahn/esp/esp-idf/components/esp_ipc/include -I/home/nthederahn/esp/esp-idf/components/app_trace/include -I/home/nthederahn/esp/esp-idf/components/esp_timer/include -I/home/nthederahn/esp/esp-idf/components/spi_flash/include -I/home/nthederahn/esp/esp-idf/components/mbedtls/port/include -I/home/nthederahn/esp/esp-idf/components/mbedtls/mbedtls/include -I/home/nthederahn/esp/esp-idf/components/mbedtls/esp_crt_bundle/include -I/home/nthederahn/esp/esp-idf/components/mdns/include -I/home/nthederahn/esp/esp-idf/components/console -I/home/nthederahn/esp/esp-idf/components/esp_adc_cal/include -I/home/nthederahn/esp/esp-idf/components/wifi_provisioning/include -I/home/nthederahn/esp/esp-idf/components/protocomm/include/common -I/home/nthederahn/esp/esp-idf/components/protocomm/include/security -I/home/nthederahn/esp/esp-idf/components/protocomm/include/transports -I/home/nthederahn/esp/esp-idf/components/nvs_flash/include -I/home/nthederahn/esp/esp-idf/components/nghttp/port/include -I/home/nthederahn/esp/esp-idf/components/nghttp/nghttp2/lib/includes -I/home/nthederahn/esp/esp-idf/components/fatfs/diskio -I/home/nthederahn/esp/esp-idf/components/fatfs/vfs -I/home/nthederahn/esp/esp-idf/components/fatfs/src -I/home/nthederahn/esp/esp-idf/components/wear_levelling/include -I/home/nthederahn/esp/esp-idf/components/sdmmc/include -I/home/nthederahn/esp/esp-idf/components/app_update/include -I/home/nthederahn/esp/esp-idf/components/bootloader_support/include -I/home/nthederahn/esp/esp-idf/components/spiffs/include -I/home/nthederahn/esp/esp-idf/components/openssl/include -I/home/nthederahn/esp/esp-idf/components/esp_hid/include -I/home/nthederahn/esp/esp-idf/components/esp_https_ota/include -I/home/nthederahn/esp/esp-idf/components/esp_http_client/include -mlongcalls -Wno-frame-address -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -ggdb -Og -fmacro-prefix-map=/mnt/c/Users/nthederahn/Projects/moscow-initializer/init-firmware=. -fmacro-prefix-map=/home/nthederahn/esp/esp-idf=IDF -fstrict-volatile-bitfields -Wno-error=unused-but-set-variable -fno-jump-tables -fno-tree-switch-conversion -std=gnu99 -Wno-old-style-declaration -D_GNU_SOURCE -DIDF_VER=\"v5.0-dev-79-gbcbef9a8db\" -DESP_PLATFORM -DARDUINO=10812 -DARDUINO_ESP32_DEV -DARDUINO_ARCH_ESP32 -DARDUINO_BOARD=\"ESP32_DEV\" -DARDUINO_VARIANT=\"esp32\" -DESP32 -MD -MT esp-idf/arduino-esp32/CMakeFiles/__idf_arduino-esp32.dir/cores/esp32/esp32-hal-timer.c.obj -MF esp-idf/arduino-esp32/CMakeFiles/__idf_arduino-esp32.dir/cores/esp32/esp32-hal-timer.c.obj.d -o esp-idf/arduino-esp32/CMakeFiles/__idf_arduino-esp32.dir/cores/esp32/esp32-hal-timer.c.obj -c ../components/arduino-esp32/cores/esp32/esp32-hal-timer.c
../components/arduino-esp32/cores/esp32/esp32-hal-timer.c: In function 'timerBegin':
../components/arduino-esp32/cores/esp32/esp32-hal-timer.c:236:16: error: 'timg_dev_t' {aka 'struct timg_dev_t'} has no member named 'int_ena'
         TIMERG1.int_ena.val &= ~BIT(timer->timer);
                ^
../components/arduino-esp32/cores/esp32/esp32-hal-timer.c:243:16: error: 'timg_dev_t' {aka 'struct timg_dev_t'} has no member named 'int_ena'
         TIMERG0.int_ena.val &= ~BIT(timer->timer);
                ^
../components/arduino-esp32/cores/esp32/esp32-hal-timer.c: In function 'timerAttachInterrupt':
../components/arduino-esp32/cores/esp32/esp32-hal-timer.c:286:20: error: 'timg_dev_t' {aka 'struct timg_dev_t'} has no member named 'int_ena'
             TIMERG1.int_ena.val &= ~BIT(timer->timer);
                    ^
../components/arduino-esp32/cores/esp32/esp32-hal-timer.c:293:20: error: 'timg_dev_t' {aka 'struct timg_dev_t'} has no member named 'int_ena'
             TIMERG0.int_ena.val &= ~BIT(timer->timer);
                    ^
../components/arduino-esp32/cores/esp32/esp32-hal-timer.c:330:20: error: 'timg_dev_t' {aka 'struct timg_dev_t'} has no member named 'int_ena'
             TIMERG1.int_ena.val |= BIT(timer->timer);
                    ^
../components/arduino-esp32/cores/esp32/esp32-hal-timer.c:336:20: error: 'timg_dev_t' {aka 'struct timg_dev_t'} has no member named 'int_ena'
             TIMERG0.int_ena.val |= BIT(timer->timer);
                    ^
[1023/1229] Building C object esp-idf/arduino-esp32/CMakeFiles/__idf_arduino-esp32.dir/cores/esp32/esp32-hal-rmt.c.obj
../components/arduino-esp32/cores/esp32/esp32-hal-rmt.c:118:25: warning: 'g_rmt_objlocks' defined but not used [-Wunused-variable]
 static xSemaphoreHandle g_rmt_objlocks[MAX_CHANNELS] = {
                         ^~~~~~~~~~~~~~
[1024/1229] Building CXX object esp-idf/arduino-esp32/CMakeFiles/__idf_arduino-esp32.dir/cores/esp32/MD5Builder.cpp.obj
ninja: build stopped: subcommand failed.
ninja failed with exit code 1
```

### PR DESCRIPTION
**Pull request by PilnyTomas**
## Summary
Updated documentation describing the usage of Arduino-esp32 core as ESP-IDF component.

## Impact
Removed confusing mentions of advanced menuconfig options.
Extended process of installation, setup, and usage.

## Related links
Closes #5833

### CODE REVIEW

File: docs/source/esp-idf_component.rst [hunk 1]
```diff
@@ -57,10 +84,10 @@ Option 1. Using Arduino setup() and loop()
 Option 2. Using ESP-IDF appmain()
 *********************************
 
-- You need to implement ``app_main()`` and call ``initArduino();`` in it.
+In main.c or main cpp you need to implement ``app_main()`` and call ``initArduino();`` in it.
```
Comment (atanisoft) [inline docs/source/esp-idf_component.rst:87]: missing . (period) between main and cpp.

Comment (PilnyTomas) [reply docs/source/esp-idf_component.rst:87]: Thank you for catching this. I will commit fix later if there is nothing else to be fixed.

File: docs/source/esp-idf_component.rst [hunk 2]
```diff
@@ -1,20 +1,30 @@
...
 
 ESP32 Arduino lib-builder
 -------------------------
 
-For a simplified method, see `lib-builder <https://github.com/espressif/esp32-arduino-lib-builder>`_.
+If you don't need any modifications in the default Arduino ESP32 core, we recommend you to install using the Boards Manager.
+
+Arduino lib builder will allow you to build your own version of Arduino core for Arduino IDE.
+
+To build your own Arduino core see `Arduino lib builder <https://github.com/espressif/esp32-arduino-lib-builder>`_
```
Comment (me-no-dev) [inline docs/source/esp-idf_component.rst:16]: It does not build the Arduino core. It builds ESP-IDF and collects all necessary headers, libraries and bootloaders. Basically everything inside `tools/sdk`.

Comment (me-no-dev) [reply docs/source/esp-idf_component.rst:16]: It is used when you need to have some specific ESP-IDF option that differs from what is already compiled in Arduino.

File: docs/source/esp-idf_component.rst [hunk 3]
```diff
@@ -69,28 +96,41 @@ If you plan to base your code on examples provided in `examples <https://github.
...
+      // WARNING: if program reaches end of function app_main() the MCU will restart.
+    }
 
 Build, flash and monitor
 ************************
 
 - For both options use command ``idf.py -p <your-board-serial-port> flash monitor``
-- It will build, upload and open serial monitor to your board.
+
+  - The port is usually ``/dev/ttyUSB0`` search the active port with ``ls /dev/ttyUSB*``
```
Comment (me-no-dev) [inline docs/source/esp-idf_component.rst:120]: this is not true on Mac or Windows, Ports have very different names. It's maybe beyond this documentation to explain how one should find the serial port.

Comment (PilnyTomas) [reply docs/source/esp-idf_component.rst:120]: Alright, I will remove this line in the next commit.

File: docs/source/esp-idf_component.rst (hunk 2 above)
Comment (PilnyTomas) [reply docs/source/esp-idf_component.rst:16]: What should the line be?

Comment (me-no-dev) [reply docs/source/esp-idf_component.rst:16]: Arduino Lib Builder is the tool that integrates ESP-IDF into Arduino. It allows you to customize the default settings used by Espressif and try them in Arduino IDE.

**Pull request merged**